src/appchooserdialog.c

```diff
@@ -341,6 +341,22 @@ app_chooser_dialog_class_init (AppChooserDialogClass *class)
   gtk_widget_class_bind_template_callback (widget_class, find_in_software);
 }
 
+static char *
+shorten_location (const char *location)
+{
+  int len;
+
+  len = g_utf8_strlen (location, -1);
+
+  if (len < 40)
+    return g_strdup (location);
+
+  for (; len >= 40; len--)
+    location = g_utf8_next_char (location);
+
+  return g_strconcat ("…", location, NULL);
```

Comment:
Not strictly necessary, but I think it would be helpful to have a comment noting that this function shortens a location by replacing the discarded substring **at the beginning** (left) of the string with a `…`. Just mentioning because I was a bit confused by the logic while reviewing it until I realized of that thing.